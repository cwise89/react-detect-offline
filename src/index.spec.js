import React from "react";
import { mount } from "enzyme";
import toJSON from "enzyme-to-json";
import { Online, Offline } from "./";

const eventListenerMap = {};
beforeEach(() => {
  window.addEventListener = jest.fn((event, cb) => {
    eventListenerMap[event] = cb;
  });
  window.removeEventListener = jest.fn(event => {
    eventListenerMap[event] = undefined;
  });
});

describe("Online", () => {
  beforeEach(() => {
    Object.defineProperty(navigator, "onLine", {
      configurable: true,
      value: true
    });
  });

  it("should render children when online", () => {
    const wrapper = mount(
      <Online>
        <h1>Hello World</h1>
      </Online>
    );

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
  
  it("should render children when online and using a custom polling URL", () => {
    const wrapper = mount(
      <Online pollingUrl="https://www.google.com/">
        <h1>Hello World</h1>
      </Online>
    );

    expect(toJSON(wrapper)).toMatchSnapshot();
  });


  it("should not render children when offline", () => {
    Object.defineProperty(navigator, "onLine", { value: false });

    const wrapper = mount(
      <Online>
        <h1>Hello World</h1>
      </Online>
    );

    expect(wrapper.html()).toBeNull();
  });
  
  it("should not render children when offline and using a custom polling URL", () => {
    Object.defineProperty(navigator, "onLine", { value: false });

    const wrapper = mount(
      <Online pollingUrl="https://www.google.com/">
        <h1>Hello World</h1>
      </Online>
    );

    expect(wrapper.html()).toBeNull();
  });


  it("should not render children when going from online to offline", () => {
    const wrapper = mount(
      <Online>
        <h1>Hello World</h1>
      </Online>
    );

    eventListenerMap.offline();

    expect(wrapper.html()).toBeNull();
  });

  it("should remove event listeners when unmounting", () => {
    const wrapper = mount(
      <Online>
        <h1>Hello World</h1>
      </Online>
    );
    wrapper.unmount();

    expect(window.removeEventListener).toHaveBeenCalledTimes(2);
    expect(window.removeEventListener.mock.calls).toMatchSnapshot();
  });
});

describe("Offline", () => {
  beforeEach(() => {
    Object.defineProperty(navigator, "onLine", {
      configurable: true,
      value: false
    });
  });

  it("should render children when offline", () => {
    const wrapper = mount(
      <Offline>
        <h1>Hello World</h1>
      </Offline>
    );

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
  
  it("should render children when offline and using a custom polling URL", () => {
    const wrapper = mount(
      <Offline pollingUrl="https://www.google.com/">
        <h1>Hello World</h1>
      </Offline>
    );

    expect(toJSON(wrapper)).toMatchSnapshot();
  });


  it("should not render children when online", () => {
    Object.defineProperty(navigator, "onLine", { value: true });

    const wrapper = mount(
      <Offline>
        <h1>Hello World</h1>
      </Offline>
    );

    expect(wrapper.html()).toBeNull();
  });
  
  it("should not render children when online and using a custom polling URL", () => {
    Object.defineProperty(navigator, "onLine", { value: true });

    const wrapper = mount(
      <Offline pollingUrl="https://www.google.com/">
        <h1>Hello World</h1>
      </Offline>
    );

    expect(wrapper.html()).toBeNull();
  });

  it("should not render children when going from offline to online", () => {
    const wrapper = mount(
      <Offline>
        <h1>Hello World</h1>
      </Offline>
    );

    eventListenerMap.online();

    expect(wrapper.html()).toBeNull();
  });

  it("should remove event listeners when unmounting", () => {
    const wrapper = mount(
      <Offline>
        <h1>Hello World</h1>
      </Offline>
    );
    wrapper.unmount();

    expect(window.removeEventListener).toHaveBeenCalledTimes(2);
    expect(window.removeEventListener.mock.calls).toMatchSnapshot();
  });
});
