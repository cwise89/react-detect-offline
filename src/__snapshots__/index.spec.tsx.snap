// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Offline should remove event listeners when unmounting 1`] = `
Array [
  Array [
    "online",
    [Function],
  ],
  Array [
    "offline",
    [Function],
  ],
]
`;

exports[`Offline should render children when offline 1`] = `
<Offline
  polling={true}
  wrapperType="span"
>
  <h1>
    Hello World
  </h1>
</Offline>
`;

exports[`Offline should render children when offline and using a custom polling URL 1`] = `
<Offline
  polling={
    Object {
      "url": "https://www.google.com/",
    }
  }
  wrapperType="span"
>
  <h1>
    Hello World
  </h1>
</Offline>
`;

exports[`Online should remove event listeners when unmounting 1`] = `
Array [
  Array [
    "online",
    [Function],
  ],
  Array [
    "offline",
    [Function],
  ],
]
`;

exports[`Online should render children when online 1`] = `
<Online
  polling={true}
  wrapperType="span"
>
  <h1>
    Hello World
  </h1>
</Online>
`;

exports[`Online should render children when online and using a custom polling URL 1`] = `
<Online
  polling={true}
  pollingUrl="https://www.google.com/"
  wrapperType="span"
>
  <h1>
    Hello World
  </h1>
</Online>
`;
